tests/unit/utils/test_utils.py

```diff
@@ -818,3 +818,26 @@ def test_libgl_workaround(monkeypatch, skip):
     if skip:
         monkeypatch.setenv('QUTE_SKIP_LIBGL_WORKAROUND', '1')
     utils.libgl_workaround()  # Just make sure it doesn't crash.
+
+
+@pytest.mark.parametrize('durations, out', [
+    ("-1s", -1),  # No sense to wait for negative seconds
+    ("-1", -1),
+    ("34ss", -1),
+    ("0", 0),
+    ("0s", 0),
+    ("59s", 59000),
+    ("60", 60000),
+    ("60.4s", -1),  # Only accept integer values
+    ("1m1s", 61000),
+    ("1m", 60000),
+    ("1h", 3_600_000),
+    ("1h1s", 3_601_000),
+    ("1s1h", 3_601_000),  # Invariant to flipping
+    ("1h1m", 3_660_000),
+    ("1h1m1s", 3_661_000),
+    ("1h1m10s", 3_670_000),
+    ("10h1m10s", 36_070_000),
+])
+def test_parse_duration(durations, out):
+    assert utils.parse_duration(durations) == out
```

Comment:
If you feel like it, it'd be great if you could also add a simple fuzzing test using [hypothesis](https://hypothesis.readthedocs.io/en/latest/index.html#). Something like:

```python
@hypothesis.given(strategies.text())
def test_parse_duration_hypothesis(duration):
    try:
        utils.parse_duration(duration)
    except ValueError:
        pass
```

that then tests that we don't accidentally crash with an unexpected error when `parse_duration` is being fed garbage.